app/models/grda_warehouse/tasks/service_history/base.rb

```diff
@@ -43,33 +43,24 @@ def process
           if @force_sequential_processing
             ::ServiceHistory::RebuildEnrollmentsJob.new(client_ids: batch, log_id: log.id).perform_now
           else
-            job = Delayed::Job.enqueue(::ServiceHistory::RebuildEnrollmentsJob.new(client_ids: batch, log_id: log.id), queue: :long_running)
+            job = Delayed::Job.enqueue(::ServiceHistory::RebuildEnrollmentsJob.new(client_ids: batch, log_id: log.id))
 
           end
         end
-        unless @force_sequential_processing
-          # Check for completion using a reasonable interval
-          interval = if @client_ids.count < 25
-            5
-          else
-            30
-          end
-          # self.class.wait_for_processing(interval: interval)
-        end
       ensure
         Rails.cache.delete('sanity_check_count')
       end
     end
 
-    def self.wait_for_processing(interval: 30, max_wait_seconds: 21_600) # 6 hours
+    def self.wait_for_processing(interval: 30, max_wait_seconds: 21_600, queue: :long_running) # 6 hours
```

Comment:
I think this queue should come from `Delayed::Job.where(queue: ::ServiceHistory::RebuildEnrollmentsByBatchJob.queue_name`